### PR TITLE
[System.Reflection] Fix the issue with constructing DllImportAttribute data

### PR DESCRIPTION
**Pull request by MaximLipnin**
Fixes https://github.com/mono/mono/issues/10544

### CODE REVIEW

File: mcs/class/corlib/System.Reflection/MonoMethod.cs [hunk 1]
```diff
@@ -455,18 +456,92 @@ internal CustomAttributeData[] GetPseudoCustomAttributesData ()
...
 
+        private CustomAttributeData GetDllImportAttributeData ()
+        {
+			if ((Attributes & MethodAttributes.PinvokeImpl) == 0)
+				return null;
+
+			string entryPoint, dllName = null;
+			PInvokeAttributes flags = 0;
+
+			((MonoMethod)this).GetPInvoke (out flags, out entryPoint, out dllName);
```
Comment (marek-safar) [inline mcs/class/corlib/System.Reflection/MonoMethod.cs]: Why do you need this cast?

File: mcs/class/corlib/System.Reflection/MonoMethod.cs [hunk 2]
```diff
@@ -455,18 +456,92 @@ internal CustomAttributeData[] GetPseudoCustomAttributesData ()
...
+
+			CharSet charSet = CharSet.None;
```
Comment (marek-safar) [inline mcs/class/corlib/System.Reflection/MonoMethod.cs]: This should be set in default case

File: mcs/class/corlib/System.Reflection/MonoMethod.cs [hunk 3]
```diff
@@ -455,18 +456,92 @@ internal CustomAttributeData[] GetPseudoCustomAttributesData ()
...
+				charSet = CharSet.Unicode; 
+				break;
+			case PInvokeAttributes.CharSetAuto: 
+				charSet = CharSet.Auto; 
+				break;
+			// Invalid: default to CharSet.None
+			default: break;
+			}
+
+			var callingConvention = InteropServicesCallingConvention.Cdecl;
```
Comment (marek-safar) [inline mcs/class/corlib/System.Reflection/MonoMethod.cs]: This should be set in default case

File: mcs/class/corlib/System.Reflection/MonoMethod.cs (hunk 1 above)
Comment (MaximLipnin) [reply mcs/class/corlib/System.Reflection/MonoMethod.cs]: No need

File: mcs/class/corlib/System.Reflection/MonoMethod.cs (hunk 2 above)
Comment (MaximLipnin) [reply mcs/class/corlib/System.Reflection/MonoMethod.cs]: done

File: mcs/class/corlib/System.Reflection/MonoMethod.cs (hunk 3 above)
Comment (MaximLipnin) [reply mcs/class/corlib/System.Reflection/MonoMethod.cs]: done

File: mcs/class/corlib/System.Reflection/MonoMethod.cs [hunk 4]
```diff
@@ -455,18 +456,96 @@ internal CustomAttributeData[] GetPseudoCustomAttributesData ()
...
+			bool preserveSig = (GetMethodImplementationFlags () & MethodImplAttributes.PreserveSig) != 0;
+
+			var ctorArgs = new CustomAttributeTypedArgument [] { 
+				new CustomAttributeTypedArgument (typeof (string), dllName),
+			};
+
+			var attrType = typeof (DllImportAttribute); 
+
+			var namedArgs = new CustomAttributeNamedArgument [] { 
+				new CustomAttributeNamedArgument (attrType.GetField ("EntryPoint"), entryPoint),
```
Comment (marek-safar) [inline mcs/class/corlib/System.Reflection/MonoMethod.cs]: Hmm, this will break with linker (will update that in separate PR)

**Pull request merged**